src/microbe_stage/Microbe.cs

```diff
@@ -904,10 +904,25 @@ public void AIThink(float delta, Random random, MicrobeAICommonData data)
     /// </returns>
     public List<(Compound Compound, Color Colour, Vector3 Target)> GetDetectedCompounds(CompoundCloudSystem clouds)
     {
+        var colonyMicrobes = Colony?.ColonyMembers ?? new List<Microbe> { this };
+        HashSet<(Compound Compound, float Range, float MinAmount, Color Colour)> colonyCompoundDetections = new();
+
+        foreach (var colonyMicrobe in colonyMicrobes)
+        {
+            foreach (var activeCompoundDetection in colonyMicrobe.activeCompoundDetections)
+            {
+                // For multiple Range/MinAmount settings this just chooses the first one
+                if (!colonyCompoundDetections.Any(o => o.Compound == activeCompoundDetection.Compound))
```

Comment:
Why do you do this? HashSet ensures already that no duplicate items are added.

Reply:
That's how it works with non-colony receptors as well. It's intentional implementation by me. So I'm 100% fine with it working that way for colonies as well.